tests/Feature/ApiTest.php

```diff
@@ -0,0 +1,512 @@
+    }
+
+    /**
+     * @depends testCreateProject
+     */
+    function testUpdateProject($id)
+    {
+        $name = rand(0, 9999999999);
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('POST', '/api/projects/' . $id, [
+            'name' => $name ,
+            'information' => 'test',
+            'deadline' => '2018-09-27',
+            'type' => 'lab',
+            'status' => 1,
+            '_method' => 'PUT'
+            ]);
+        $response->assertStatus(200)
+        ->assertJson(['status' => true, 'message' => 'Product Updated Successfully.']);
+    }
+
+    /**
+    * @depends testCreateMember
+    * @depends testCreateProject
+    */
+    function assign($m_id, $p_id)
+    {
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('POST', '/api/project/assign',['project_id' => $p_id,
+            'member_id' => $m_id,
+            'role' => 'dev'
+            ]);
+        $response->assertStatus(200)
+        ->assertJson([
+            'status' => true,
+            'message' => 'Assign Successfully.'
+            ]);
+    }
+
+    /**
+     * @depends testCreateMember
+     * @depends testCreateProject
+     */
+    function unassign($m_id, $p_id)
+    {
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('DELETE', '/api/project/detach',['project_id' => $p_id,'member_id' => $m_id]);
+        $response->assertStatus(200)
+        ->assertJson([
+            'status' => true,
+            'message' => 'Unassign Successfully.'
+            ]);
+    }
+
+    /**
+     * @depends testCreateMember
+     */
+    function deleteMember($id)
+    {
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('DELETE', '/api/members/' . $id);
+        $response->assertStatus(200)
+        ->assertJson([
+            'status' => true,
+            'message' => 'Member Deleted Successfully.'
+            ]);
+    }
+
+    /**
+     * @depends testCreateProject
+     */
+    function testDeleteProject($id)
+    {
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('DELETE', '/api/projects/' . $id);
+        $response->assertStatus(200)
+        ->assertJson([
+            'status' => true,
+            'message' => 'Project Deleted Successfully.'
+            ]);
+    }
+
+    function testUploadFile()
+    {
+        $name = time();
+        $response = $this->withHeaders(['X-Requested-With', 'XMLHttpRequest'])
+        ->json('POST', '/api/members', [
+            'name' => 'upload' . $name ,
+            'information' => '1234567891',
+            'dob' => '1992-09-27',
+            'position' => 'intern',
+            'phone' => '0965841492',
+            'gender' => 2,
+            'avatar' =>  new UploadedFile('/home/sunshine/Downloads/thank_you_3.jpg', 'thank_you_3.jpg', "image/jpg", 70, null, true)
```

Comment:
You should add this file to the repository and load it from a relative path.

Reply:
thanks